Marlin/src/pins/stm32/pins_LONGER3D_LK.h

```diff
@@ -138,21 +142,28 @@
   #define TOUCH_INT_PIN    PC6   // pin 63 (PenIRQ coming from ADS7843)
 
   #define BTN_ENC          PB0   // pin 35 unconnected pin on Alfawise. (PC13 to try)
-  #define BTN_EN1          -1    // Real pin is needed to enable encoder's push button
-  #define BTN_EN2          -1    // functionality used by touch screen
 #endif
 
 //
-// SPI1 (EEPROM W25Q64 + DAC OUT)
+// Persistent Storage
+// If no option is selected below the SD Card will be used
 //
+//#define SPI_EEPROM
+#define FLASH_EEPROM_EMULATION
+
+#if ENABLED(SPI_EEPROM)
+  // SPI1 (EEPROM W25Q64 + DAC OUT)
+  #define SPI_CHAN_EEPROM1   1
+  #define SPI_EEPROM1_CS     PC5   // pin 34
+  #define EEPROM_PAGE_SIZE   0x1000U   // 4KB
+  #define EEPROM_SCK  BOARD_SPI1_SCK_PIN    // PA5 pin 30
+  #define EEPROM_MISO BOARD_SPI1_MISO_PIN   // PA6 pin 31
+  #define EEPROM_MOSI BOARD_SPI1_MOSI_PIN   // PA7 pin 32
+#elif ENABLED(FLASH_EEPROM_EMULATION)
+  // SoC Flash (framework-arduinoststm32-maple/STM32F1/libraries/EEPROM/EEPROM.h)
+  #define EEPROM_PAGE_SIZE     0x800U
+  #define EEPROM_START_ADDRESS (0x8000000UL + (512 * 1024) - 2 * EEPROM_PAGE_SIZE)
+#endif
+
 #undef E2END
-#define E2END              0x7FF // EEPROM end address (reserve 2kB on sd/sram, real spi one is 8MB/64Mbits)
-/*
-#define SPI_EEPROM         1   // If commented this will create a file on the SD card as a replacement
-#define SPI_CHAN_EEPROM1   1
-#define SPI_EEPROM1_CS     PC5 // pin 34
-
-//#define EEPROM_SCK  BOARD_SPI1_SCK_PIN   // PA5 pin 30
-//#define EEPROM_MISO BOARD_SPI1_MISO_PIN  // PA6 pin 31
-//#define EEPROM_MOSI BOARD_SPI1_MOSI_PIN  // PA7 pin 32
-*/
+#define E2END (EEPROM_PAGE_SIZE - 1)
```

Comment:
may cause issues on SD, using EEPROM_PAGE_SIZE

Reply:
Right. So, I need a default case to set it if it is not defined.

Reply:
euh it was :p 0x7FF